imap/jmap_api.h

```diff
@@ -191,7 +191,9 @@ extern void jmap_finireq(jmap_req_t *req);
 
 extern int jmap_is_using(jmap_req_t *req, const char *capa);
 
-#define JMAP_SHARED_CSTATE 1 << 0
+#define JMAP_READ_ONLY     (1 << 0)
```

Comment:
I find this READ_ONLY flag confusing. I guess it means that methods flagged with this are safe to call in a read-only instance? A possible source of subtle errors is that nothing really enforces the read-only part, in contrast to a shared cstate which will certainly bail if the method implementation attempts to create messages or the like. I guess I wonder if we can enforce READ_ONLY on the lower levels, such that no annotations, mailboxes, messages etc can be altered at all.